### ISSUE TITLE
Lambda cross-region role chaining errors

### ISSUE THREAD

**Issue by jaymiracola** [solo-io/gloo#8578]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.14.9

### Kubernetes Version

1.25

### Describe the bug

After setting up role chaining per docs, and requesting the US, the following error is observed . `no credentials present for AWS upstream`

The error is not seen when referencing the original role.

### Expected Behavior

Role chaining should not produce an error. 

### Steps to reproduce the bug

1. Role 1 with policy to access Lambda. Lambda and EKS live in us-east-2
2. Role 2 referencing role 1
3. Us configured as
```
apiVersion: gloo.solo.io/v1
kind: Upstream
metadata:
  name: aws-upstream
  namespace: gloo-system
spec:
  aws:
    lambdaFunctions:
    - lambdaFunctionName: hello
      logicalName: hello
    region: us-east-2
    roleArn: arn:aws:iam::<account-num>:role/jm-lambda-b
```

**Comment by jaymiracola** [comment]
@nfuden @ben-taussig-solo

**Comment by ben-taussig-solo** [comment]
I created a debug gloo-envoy-wrapper image from the assets built during this PR's CI run: https://github.com/solo-io/envoy-gloo/pull/266

This PR hardcoded the region used to sign the request to assume the secondary role to `us-east-2`. Jay tested this image in the environment he used to replicate the bug and continued to see the same behavior.

**Comment by ben-taussig-solo** [comment]
One of my PRs must have not set `resolvesIssue: false` -- reopening the issue until all PRs have merged

**Comment by ben-taussig-solo** [comment]
This is set to be released in Gloo EE v1.15.3, v1.14.10, and v1.13.24. Closing the issue now
